ci/build_and_test.sh

```diff
@@ -18,6 +18,12 @@ fi
 cargo test
 
 # Build and test other crates
-cd config_proc_macro
-cargo build --locked
-cargo test
+crates=(config_proc_macro check_diff)
+for crate in "${crates[@]}"
+do
+    cd $crate
+    cargo build --locked
+    cargo test
+    # change back to root
+    cd ..
+done
```

Comment:
To keep things consistent between both scripts lets revert this and apply the suggested change from https://github.com/rust-lang/rustfmt/pull/6187/files#r1678741100 but with bash syntax.